File: roman.rb
def old_roman_numeral(num)
  refer = { "M" => 1000, "D" => 500, "C" => 100, "L" => 50, "X" => 10, 
            "V" => 5, "I" => 1 }
  numeral = ""

  until num == 0
    refer.each do |key, value|
      next if value > num
      numeral += key
      num = num - value
      break
    end
  end
  numeral
end


p old_roman_numeral(3755) # MMMDCCLV

File: roman.js
const oldRomanNumeral = (num) => {
  const numerals = ['M', 'D', 'C', 'L','X', 'V', 'I']
  const numbers  = [1000, 500, 100, 50, 10, 5, 1]
  let roman = ''
  let i = 0

  while(num > 0){
    numbers.forEach(number => {
      if(numbers[i] > num){
        return
      } else {
      roman += numerals[i]
      num = num - numbers[i]
      }
    })
    i++
  }
  return roman
}


console.log(oldRomanNumeral(3755)) // MMMDCCLV
